utils.js

```diff
@@ -0,0 +1,31 @@
+import {fileURLToPath} from 'node:url';
+import {Transform} from 'node:stream';
+
+export const toPath = urlOrPath => {
+	if (!urlOrPath) {
+		return urlOrPath;
+	}
+
+	if (urlOrPath instanceof URL) {
+		urlOrPath = urlOrPath.href;
+	}
+
+	return urlOrPath.startsWith('file://') ? fileURLToPath(urlOrPath) : urlOrPath;
```

Comment:
I just realized, we shouldn't do `urlOrPath.startsWith('file://')`. Only the instanceof check. We should only accept `URL` or string path.